R/annotate_sv.R

```diff
@@ -2,7 +2,7 @@
 #'
 #' @description Annotate a data frame of SV breakpoints represented in an extended BEDPE format.
 #'
-#' @details Specify a data frame with SVs (preferably the output from [GAMBLR::get_manta_sv]) to the `sv_df` parameter and get back the same data frame with SV annotations.
+#' @details Specify a data frame with SVs (preferably the output from [GAMBLR.results::get_manta_sv], if the user has GSC-restricted access, or `GAMBLR.data::sample_data$grch37$bedpe` or `GAMBLR.data::sample_data$hg38$bedpe` otherwise) to the `sv_df` parameter and get back the same data frame with SV annotations.
```

Comment:
In the overhaul of GAMBLR.viz I never added any hyperlinks to `GAMBLR.results` functions. I did this because I wasn't sure how that would affect the dependency structure. Instead, I just typed the function name and referred to it being available in the GAMBLR.results package. So something like this;

> Specify a data frame with SVs (preferably the output from `get_manta_sv` in GAMBLR.results package. If the user has GSC-restricted access one of the following BEDPEs are available in the. GAMBLR.data package; `GAMBLR.data::sample_data$grch37$bedpe`, `GAMBLR.data::sample_data$hg38$bedpe`) to the `sv_df` parameter and get back the...

Reply:
The link `[packageA::function1]` doesn't need `packageA` as a dependency. If `packageA` is installed, the link works, otherwise the link leads to something like `Internal Server Error`. However, in both cases, the text `packageA::function1` will be shown correctly in the documentation. GAMBLR.results (main branch) is a good example to see this working (just keep in mind that you need to install the package; `devtools::load_all()` wouldn't be enough).

Reply:
Just want to clarify here:
1. If the user does not have the GAMBLR.results installed, the link like this will show the `Internal Server Error`
2. If the user does have the GAMBLR.results installed, the help message will properly show up

If the 1 is true, we should strip all of the mentions of the GAMBLR.results from the documentation.

Reply:
> If the user does not have the GAMBLR.results installed, the link like this will show the Internal Server Error

Yes, this is true. Now that I have my GAMBLR.results uninstalled, this is what I get when I click on the `GAMBLR.results::get_manta_sv` link:
![Screenshot from 2023-09-05 15-25-58](https://github.com/morinlab/GAMBLR.utils/assets/22369549/acb89305-c529-4a4b-9e84-47338f4ebc15)

But I don't see this as a problem, rather as expected --- the documentation states that the user needs GSC-restricted access. I might be wrong, but if we keep only examples for non-GSC users, and don't tell GSC users what they should/could be doing, I just feel that the documentation is not complete when we could have it completed. But If you really think that we should revert this change to what it was, I would be fine with doing it though.

Reply:
I agree with Kostia on this. If anything, errors should be avoided as much as possible. Imagine a non-GSC user running GAMBLR, I can easily see that an Internal Server Error can be confusing. Especially if you imagine that a user of this group would probably have a limited understanding of the structure design of these child repos. But this is just my opinion and I would be fine with leaving it as is. In the future, and if we think (and based on user feedback) this change is required, it's an easy implementation.

Reply:
All right. I see your point. Thanks. I liked the idea that you showed here, and I'm going to follow it: 
> "... preferably the output from `get_manta_sv` in GAMBLR.results package ..."

One question, please. How will the documentation of the GAMBLR.data and GAMBLR.results be? Will they be the same? Will they mention that there is another version and the version to use is chosen automatically?

Reply:
This is something I have not thought about in greater detail. But from the top of my head, I think it would no harm to inform the user that there are two versions available for a given function and the specific usage depends on what data permission the user has set up. For example in `GAMBLR.data::get_manta_sv` it would make sense to let the user know that there is another version of this function with additional parameters available in the `GAMBLR.results` package. But this is something we can discuss in greater detail once everything is set up and we're tackling documentation, vignettes, etc. Thanks!

Reply:
I removed the link. Now, no documentation contains links to GAMBLR.results.